src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendMerger.java

```diff
@@ -343,7 +344,7 @@ public static BufferedImage mergeGroups(Rule[] rules, MergeOptions mergeOptions)
         }
 
         if (layout == LegendLayout.VERTICAL) {
-            Column[] columns = createColumns(nodes, 0, 0);
+            Column[] columns = createColumns(nodes, 0, 0, null, false);

```

Comment:
Do you know when this gets called, and why it doesn't need to truncate the legend?